flink-scala/src/main/scala/org/apache/flink/api/scala/typeutils/OptionSerializer.scala

```diff
@@ -65,7 +65,7 @@ class OptionSerializer[A](val elemSerializer: TypeSerializer[A])
     case Some(a) =>
       target.writeBoolean(true)
       elemSerializer.serialize(a, target)
-    case None =>
+    case _ =>
```

Comment:
Sure, we have to change `deserialize()` as well. 
If we can check for `2` (i.e, `null`) before we check for `1` (`SOME`), it should work, no?